docs/REST/authentication.md

```diff
@@ -530,4 +530,43 @@ $secret = "<Secret>"
 
 $headers = Sign-Request $uri.Authority $method $uri.PathAndQuery $body $credential $secret
 Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -Body $body
+```
+### Bash
+```Bash
+#!/bin/bash
+
+sign_request () {
+    local host="$1"
+    local method="$2"      # GET, PUT, POST, DELETE
+    local url="$3"         # path+query
+    local body="$4"        # request body
+    local credential="$5"  # access key id
+    local secret="$6"      # access key value (base64 encoded)
+
+    local verb="${method^^}"
+    local utc_now="$(date -u '+%a, %d %b %Y %H:%M:%S GMT')"
+    local content_hash="$(printf "$body" | openssl sha256 -binary | base64)"
+
+    local signed_headers="x-ms-date;host;x-ms-content-sha256"  # Semicolon separated header names
+    local string_to_sign="$verb\n$url\n$utc_now;$host;$content_hash"  # Semicolon separated signed_headers values
+
+    local decoded_secret="$(printf "$secret" | base64 -d)"
+    local signature="$(printf "$string_to_sign" | openssl sha256 -hmac "$decoded_secret" -binary | base64)"
+
+    # Output request headers
+    printf '%s\n' \
+           "x-ms-date: $utc_now" \
+           "x-ms-content-sha256: $content_hash" \
+           "Authorization: HMAC-SHA256 Credential=$credential&SignedHeaders=$signed_headers&Signature=$signature"
+}
+
+host="{config store name}.azconfig.io"
+method="GET"
+url="/kv"
+body=""
+credential="<Credential>"
+secret="<Secret>"
+
+mapfile -t headers <<< $(sign_request "$host" "$method" "$url" "$body" "$credential" "$secret")
+curl -X "$method" -d "$body" "${headers[@]/#/-H}" "https://$host$url"
```

Comment:
>curl [](start = 0, length = 4)

I try this at Bash on Windows and I still get 401, please check this working cross platforms

Reply:
I'm seeing this work using Bash on Windows (Ubuntu 18.04.2 LTS) - Which version are you using?

Reply:
GNU bash, version 4.4.23(1)-release-(x86_64-pc-msys)
OpenSSL 1.1.1a  20 Nov 2018
curl 7.55.1 (Windows) libcurl/7.55.1 WinSSL


---
In reply to: [333760782](https://github.com/Azure/AppConfiguration/pull/159#discussion_r333760782) [](ancestors = 333760782)

Reply:
I obtained this version of bash from the msys2 distro, but it is working for me.  Could you try again with the current version of the sample?  If it doesn't work for you, could you send me the output when running bash with `-x`?

Reply:
I sent you email about the output with -x

---
In reply to: [340255522](https://github.com/Azure/AppConfiguration/pull/159#discussion_r340255522) [](ancestors = 340255522)

Reply:
Thanks, I've made a change to the openssl call - Can you try again with the latest version?

Reply:
Resolved offline and updated prerequisites section